### PR TITLE
Some xenomorph ability PR.

### PR DESCRIPTION
**Pull request by Ghommie**
Reprisal of that one PR regarding xeno's ability to absorb plasma from other dead xenos.

:cl: Ghommie
add: "Salvage plasma", A new xeno ability for drones and old admin spawn xenoborgs has been added. It works similarly to the inverse of "Transfer plasma", but requires the donor to be dead.
bugfix: Carbon xenos can no longer transfer plasma to silicons' ones and viceversa.
tweak: Improved xeno plasma transfer ability's messages a little.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/life.dm [hunk 1]
```diff
@@ -11,6 +11,9 @@
 	..()
 
 	if(stat == DEAD) //Dead, nothing else to do.
+		if(stat == DEAD) //Dead, nothing else to do but this.
```
Comment (DominikPanic) [inline code/modules/mob/living/carbon/xenomorph/life.dm]: `if(stat == DEAD)` is here twice.

File: code/modules/mob/living/carbon/xenomorph/Powers.dm [hunk 2]
```diff
@@ -982,30 +982,34 @@
...
+	if(!isXeno(A) || !check_state())
 		return
+
 	var/mob/living/carbon/Xenomorph/target = A
+	var/is_silicon = xeno_caste & CASTE_IS_ROBOTIC ? TRUE : FALSE
+	var/energy = is_silicon ? "power" : "plasma"
 
-	if(!check_state())
+	if(!isturf(loc))
+		to_chat(src, "<span class='warning'>You can't transfer  [energy] from here!</span>")
```
Comment (DominikPanic) [inline code/modules/mob/living/carbon/xenomorph/Powers.dm]: `transfer  [energy]` double space here.

File: code/modules/mob/living/carbon/xenomorph/Castes/Queen.dm [hunk 3]
```diff
@@ -46,6 +46,10 @@
 	// *** Queen Abilities *** //
 	queen_leader_limit = 1 //Amount of leaders allowed
 
+/datum/xeno_caste/predalien/handle_decay()
```
Comment (DominikPanic) [inline code/modules/mob/living/carbon/xenomorph/Castes/Queen.dm]: Are you sure you want to handle the predalien here?

File: code/modules/mob/living/carbon/xenomorph/Powers.dm [hunk 4]
```diff
@@ -1020,10 +1024,70 @@
...
+		to_chat(src, "<span class='warning'>You need to be closer to [target].</span>")
+		return
+
+	if(!(target.plasma_stored))
+		to_chat(src, "<span class='notice'>[target] doesn't have any [energy] left to salvage.</span>")
+		return
+
+	to_chat(src, "<span class='notice'>You start salvaging [energy] from [target].</span>")
+
+	while(target.plasma_stored)
```
Comment (DominikPanic) [inline code/modules/mob/living/carbon/xenomorph/Powers.dm]: Are you sure a while loop is the best way to do this? When those loops go wrong, they usually go /very/ wrong. Wouldn't it better so xenos have to salvage multiple times so no looping would be neccessary?

File: code/modules/mob/living/carbon/xenomorph/Castes/Larva.dm [hunk 5]
```diff
@@ -25,7 +25,7 @@
 	evolves_to = list(/mob/living/carbon/Xenomorph/Drone, /mob/living/carbon/Xenomorph/Runner, /mob/living/carbon/Xenomorph/Sentinel, /mob/living/carbon/Xenomorph/Defender)
 
 	// *** Flags *** //
-	caste_flags = CASTE_CAN_BE_QUEEN_HEALED|CASTE_EVOLUTION_ALLOWED|CASTE_CAN_VENT_CRAWL|CASTE_INNATE_HEALING
+	caste_flags = CASTE_CAN_BE_QUEEN_HEALED|CASTE_EVOLUTION_ALLOWED|CASTE_CAN_VENT_CRAWL|CASTE_INNATE_HEALING|CASTE_NO_DECAY
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/xenomorph/Castes/Larva.dm]: Can you avoid negative flags? That way we avoid double-negatives on checks. Something like `CASTE_ENERGY_PLASMA` or `CASTE_SILICON` would be better.

**Comment by Rohesie** [comment]
Also conflicts.

File: code/modules/mob/living/carbon/xenomorph/Castes/Larva.dm (hunk 5 above)
Comment (Ghommie) [reply code/modules/mob/living/carbon/xenomorph/Castes/Larva.dm]: Will `CASTE_DECAY_PROOF` suffice?

Comment (Rohesie) [reply code/modules/mob/living/carbon/xenomorph/Castes/Larva.dm]: Works.

**Comment by Rohesie** [comment]
On the changelog, is this backend only?
The first tweak is actually a bugfix.

File: .travis.yml [hunk 6]
```diff
@@ -27,5 +27,5 @@ before_script:
   - tools/travis/before_build_byond.sh
 
 script:
-  - tools/travis/check_filedirs.sh ColonialMarinesALPHA.dme
```
Comment (spookydonut) [inline .travis.yml]: different PR for this please

File: code/__DEFINES/__game.dm [hunk 7]
```diff
@@ -81,22 +81,23 @@ These are used with cdel (clean delete). For example, qdel(atom, TA_REVIVE_ME) w
...
-#define CHAT_GHOSTHIVEMIND	8192
+#define CHAT_OOC			(1 << 0)
+#define CHAT_DEAD			(1 << 1)
+#define CHAT_GHOSTEARS		(1 << 2)
+#define CHAT_GHOSTSIGHT		(1 << 3)
+#define CHAT_PRAYER			(1 << 4)
+#define CHAT_RADIO			(1 << 5)
+#define CHAT_ATTACKLOGS		(1 << 6)
+#define CHAT_DEBUGLOGS		(1 << 7)
+#define CHAT_GHOSTRADIO 	(1 << 8)
```
Comment (spookydonut) [inline code/__DEFINES/__game.dm]: this shouldnt be in this PR

File: code/__DEFINES/__game.dm [hunk 8]
```diff
@@ -81,22 +81,23 @@ These are used with cdel (clean delete). For example, qdel(atom, TA_REVIVE_ME) w
...
+#define SHOW_TYPING 		(1 << 9)
+#define CHAT_FFATTACKLOGS 	(1 << 10)
+#define CHAT_ENDROUNDLOGS	(1 << 11)
+#define CHAT_GHOSTHIVEMIND	(1 << 12)
+#define CHAT_STATISTICS		(1 << 13)
```
Comment (spookydonut) [inline code/__DEFINES/__game.dm]: definitely should be in a different PR

File: code/game/gamemodes/cm_process.dm [hunk 9]
```diff
@@ -170,6 +170,41 @@ dat += " You failed to evacuate \the [MAIN_SHIP_NAME]"
 		to_chat(world, dat)
 
 
+/datum/game_mode/proc/declare_completion_announce_round_stats()
```
Comment (spookydonut) [inline code/game/gamemodes/cm_process.dm]: this has nothing to do with xenomorph abilities, please split it into a separate PR

File: .travis.yml (hunk 6 above)
Comment (Ghommie) [reply .travis.yml]: It's some merge garbage which crawled in here. Pretty much already merged stuff from different PRs.
Perhaps it may be because I didn't properly name the conflict resolving commit, but may have no impact on the actual code afaik.

Comment (spookydonut) [reply .travis.yml]: still here
